cogs5e/lookup.py

```diff
@@ -730,55 +559,69 @@ async def _get_destination(self, ctx):
         pm = guild_settings.get("pm_result", False)
         return ctx.author if pm else ctx.channel
 
-    async def _lookup_search(self, ctx, choices, query, key, search_type=None, is_obj=False):
-        if is_obj:
-            selectkey = self.nsrd_selectkey_obj
-        else:
-            selectkey = self.nsrd_selectkey
+    async def _lookup_search3(self, ctx, entities, query, query_type=None):
```

Comment:
This is the main function that interacts with entitlements in lookup - it takes a Context (which includes the Discord user), a map of {entity_type: entities}, and the user query.

For each entity type in the map, it queries `BeyondClient.get_accessible_entities()` to determine which entities are visible to the user. If the user selects an entity they cannot view, it raises a `RequiresLicense` (handled in gamedata.lookuputils).